File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    # 여벌의 체육복이 있으면서 도난당한 학생들을 제거하고, 빌려줄 수 있는 학생 수를 계산
    common_students = set(lost) & set(reserve)
    lost = list(set(lost) - common_students)
    reserve = list(set(reserve) - common_students)
    answer = n - len(lost)

    # 도난당한 학생에게 체육복 빌려주기
    for r in reserve:
        if r - 1 in lost:
            answer += 1
            lost.remove(r - 1)
        elif r + 1 in lost:
            answer += 1
            lost.remove(r + 1)

    return answer

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, initLost, initReserve) {
    var answer = 0;
    const lostSet = new Set(initLost);
    const reserveSet = new Set(initReserve);
    
    const lost = Array.from(new Set([...lostSet].filter(x => !reserveSet.has(x))));
    const reserve = Array.from(new Set([...reserveSet].filter(x => !lostSet.has(x))));
    
    console.log('lost:',lost)
    console.log('reserve:',reserve)
    
    let students = Array(n).fill(1)
    // 왼쪽에서부터 점검해주면 된다.
    for (idx = 0; idx < lost.length; idx++) {
        console.log('d1',idx);
        students[lost[idx] - 1] -= 1
    }
    for (idx = 0; idx < reserve.length; idx ++) {
        console.log('d2',idx);
        students[reserve[idx] - 1] += 1
    }
    console.log(students)
    
    for (idx = 0; idx < n; idx++) {
        if (students[idx] == 0) {
            if (idx > 0 && students[idx - 1] >= 2) {
                students[idx] += 1
                students[idx - 1] -= 1
            } else if (idx < n - 1 && students[idx + 1] >= 2) {
                students[idx] += 1
                students[idx + 1] -= 1
            }
        }
    }
    console.log(students)
    for (idx = 0; idx < n; idx++) {
        if (students[idx]) {
            answer += 1
        }
    }
    return answer;
}